### ISSUE TITLE
Allow to query on read-only file system

### ISSUE THREAD

**Issue by BLLuis** [rakibtg/SleekDB#67]
Sometimes I need to run some scripts without write permissions on file system and SleekDB throws an exception about that.
So, consider to add a new flag for readonly operations (something like the auto_cache flag), in this case, let the code throw the exception on file_put_contents (for example).

A copy of exception:

```
PHP Fatal error:  Uncaught Exception: Data directory is not writable at "/home/blah/dataDB/." Please change data directory permission. in /home/blah/SleekDB/traits/helpers.php:27
Stack trace:
#0 /home/blah/php/SleekDB/SleekDB.php(20): SleekDB\SleekDB->init(Array)
#1 /home/blah/php/SleekDB/SleekDB.php(26): SleekDB\SleekDB->__construct('/home/blah...', Array)
#2 /home/blah/php/demo.php(22): SleekDB\SleekDB::store('testLog', '/home/blah...', Array)
#3 {main}
  thrown in /home/blah/php/SleekDB/traits/helpers.php on line 27
```

Regards.

**Comment by rakibtg** [comment]
@Timu57 Can you handle this in the refactored version?

**Comment by Timu57** [comment]
Yes I can :+1:

**Comment by Timu57** [comment]
Solved in #68

Now SleekDB just checks the permissions it needs to execute a task. That means read only is possible if caching is disabled.
